esinstall/src/rollup-plugins/rollup-plugin-strip-source-mapping.ts

```diff
@@ -9,7 +9,7 @@ export function rollupPluginStripSourceMapping(): Plugin {
   return {
     name: 'snowpack:rollup-plugin-strip-source-mapping',
     transform: (code) => ({
-      code: code.replace(/[^'"`]\/\/+#\s*sourceMappingURL=.+$/gm, ''),
+      code: code.replace(/\/\/#\s*sourceMappingURL=.+$/, ''),
```

Comment:
@FredKSchott this now breaks the “mid-file“ rule; maybe we should limit to newline-only instead? I think that’d be muuuch safer

Reply:
Yea, unfortunately we see this mid-file after bundling multiple files together. That would be a regression (I think lit-html or lit-element was the most popular known example of this).

I know you hate this, but these are such one-off weird cases I'd rather handle this as a one-off solution. How about:


```diff
- code.replace(/[^'"`]\/\/+#\s*sourceMappingURL=.+$/gm, '')
+ code.replace(/[^'"`]\/\/+#\s*sourceMappingURL=[^'"`].+$/gm, '')
```

Reply:
What breaks it is the `\n` too so it‘d have to be:

```
/[^'"`](\\n)?\/\/+#\s*sourceMappingURL=[^'"`]+$/gm
```

If it works it works, but I‘m gonna add some tests for this plugin specifically to make sure

Reply:
👍  sounds good

Reply:
Actually, after adding tests, there are more problems with the current RegEx:

![Screen Shot 2020-11-06 at 11 03 57 AM](https://user-images.githubusercontent.com/1369770/98399592-dedc8100-201f-11eb-9876-3e9195d3d0f8.png)

Reply:
It seems to strip any character that comes before it. Gonna try and get these tests passing, then you can tell me how you feel about the result